src/stream/src/executor/mod.rs

```diff
@@ -182,6 +182,24 @@ pub struct ExecutorInfo {
 
     /// Identity of the executor.
     pub identity: String,
+
+    /// The executor id of the executor.
+    pub id: u64,
+}
+
+impl ExecutorInfo {
+    pub fn new(schema: Schema, pk_indices: PkIndices, identity: String, id: u64) -> Self {
+        Self {
+            schema,
+            pk_indices,
+            identity,
+            id,
+        }
+    }
+
+    pub fn new_for_test(schema: Schema, pk_indices: PkIndices, identity: String) -> Self {
+        Self::new(schema, pk_indices, identity, 0)
+    }
```

Comment:
Why not include `id` for this method? In most cases, `identity` is like `XxxExecutor {id}`. I think we can actually accept `name: String, id: u64` and format it for the caller.

Reply:
Sure we can accept it. This keeps the same behaviour as before. If you look at the test code, they construct the executor without id.

Reply:
Updated https://github.com/risingwavelabs/risingwave/pull/21165/commits/8c38b2038d937c069750dc3effde1e510ff23c42